File: Module3/mfragments36/app/src/main/java/ru/myitschool/mte/MainActivity.java
package ru.myitschool.mte;

import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import ru.myitschool.mte.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private Handler handler;
    private Boolean isRunning = false;
    private final Runnable runnable = new Runnable() {
        @Override
        public void run() {
            if (isRunning) {
                handler.postDelayed(this, 3000);
                toggleFragments();
            }
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ru.myitschool.mte.databinding.ActivityMainBinding binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        handler = new Handler(Looper.getMainLooper());

        if (savedInstanceState == null)
            getSupportFragmentManager().beginTransaction()
                    .add(R.id.output_fragment, new FirstFragment())
                    .commit();

        Button btnStart = binding.content.startBtn;
        Button btnStop = binding.content.stopBtn;

        btnStart.setOnClickListener(view -> {
            if (!isRunning) {
                isRunning = true;
                handler.post(runnable);
            }
        });

        btnStop.setOnClickListener(view -> {
            isRunning = false;
            handler.removeCallbacks(runnable);
        });
    }


    @Override
    protected void onPause() {
        super.onPause();
        handler.removeCallbacks(runnable);
    }

    private void toggleFragments() {
        Fragment fragment = (getSupportFragmentManager().findFragmentById(R.id.output_fragment) instanceof FirstFragment) ? new ProceedingFragment() : new FirstFragment();
        getSupportFragmentManager().beginTransaction()
                .replace(R.id.output_fragment, fragment)
                .commit();
    }
}

File: Module3/PR3.5/app/src/main/java/ru/adwantay/lab/MainActivity.kt
package ru.adwantay.lab

import android.os.Bundle
import android.util.Log
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.lifecycleScope

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction().apply {
                add(R.id.fragment1, RedFragment())
                add(R.id.fragment2, BlueFragment())
            }.commit()
        }

        findViewById<Button>(R.id.caption).setOnClickListener {
            switchFragments()
        }

    }

    private fun switchFragments() {
        val fragment1 =
            if (supportFragmentManager.findFragmentById(R.id.fragment1) is BlueFragment) RedFragment() else BlueFragment()
        val fragment2 = if (fragment1 is BlueFragment) RedFragment() else BlueFragment()

        supportFragmentManager.beginTransaction().apply {
            replace(R.id.fragment1, fragment1)
            replace(R.id.fragment2, fragment2)
        }.commit()
    }


}